### ISSUE TITLE
[Deprecation]: Automatically apply deprecation when new versions are uploaded

### ISSUE THREAD

**Issue by loic-sharma** [NuGet/NuGetGallery#8872]
### Related Problem

Imagine my package `Old` has been replaced by `New`. I still support `Old` and upload new versions to fix security issues. Since I want all my customers to move to `New`, I deprecate all versions of `Old`.

Unfortunately, package deprecations only apply to existing versions: a newly uploaded version will not be marked as deprecated. I need to remember to login to nuget.org and deprecate each version after I upload it.

### The Elevator Pitch

There should be a way to automatically apply a deprecation all versions of a package, including future uploaded versions.

### Additional Context and Details

This is an ask from the Azure SDK team: https://www.nuget.org/packages/Microsoft.Azure.DocumentDB/

/cc @chgill-MSFT @petermarcu

**Comment by kirankumarkolli** [comment]
Cosmos DB V2 SDK's were deprecated. Even post deprecation we are bound to address critical security and bug fixes for existing customers. We uncovered that those updated packages are not not marked deprecated and its a manual process which is fragile and error prone. Its a fundamental requirement which should block accidental mis-interpretations  and unintended dependencies.

**Comment by lfraleigh** [comment]
This just impacted the ServiceBus package as well.  They had to release a bug fix and that release was not marked as deprecated automatically. https://www.nuget.org/packages/Microsoft.Azure.ServiceBus/5.2.0

**Comment by chgill-MSFT** [comment]
I would really like this experience to be exposed in the in UI. @loic-sharma can validate these with numbers, but I have the following hypotheses:

1. When using the deprecation flow, most users want to deprecate all versions of their package, not a specific subset of versions. In other words, the most common goal with deprecation is to deprecate on the id level, not version level.
2. When using the deprecation flow to deprecate all versions of a package, users can get confused and accidentally only deprecate the latest version of the package since it's the default selection.
3. When pushing patches to deprecated packages, users forget to deprecate the newly uploaded patch which essentially removes deprecation indicators in search. (The original problem that led to this issue)

My proposal is that we should enable a clear path to deprecating a package on the id level. Doing so should mark all existing versions as deprecated and automatically mark future versions are deprecated.

Here are a couple ideas:

**Idea 1**
There is selection box outside the version selector that disabled the version selector dropdown.
![image](https://user-images.githubusercontent.com/15097183/142699806-2c959724-2f0a-47e8-b457-b3107e3bbc42.png)

**Idea 2**
There are clear and separate paths for deprecating all current and future versions vs deprecating a subset.
![image](https://user-images.githubusercontent.com/15097183/142699985-be6b3943-51e2-4e43-ae88-1d26d84e9f40.png)

**Comment by Petermarcu** [comment]
I agree with what you are saying @chgill-MSFT ! As @lfraleigh mentioned above, we just had another partner team get bit by this. It would be good to get a near term solution in place to stop the bleeding and then get it wired into all the right UX and defaults.

**Comment by loic-sharma** [comment]
Thanks for the feedback @Petermarcu! We want to make sure we understand the "north star" user experience so that our engineering MVP aligns to that experience. Like you said, we plan to first create a near term solution to unblock our partners. Do please let us know if you have concerns with the "north star" UX as that may affect engineering decisions!

**Comment by scottaddie** [comment]
The ASP.NET team also hit this problem recently with the [Microsoft.Extensions.Configuration.AzureKeyVault](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.AzureKeyVault) package. Tagging @Pilchie, as I'm sure he has feedback on how this feature should work.
